### PR TITLE
OpenAPI and datetime updates

### PR DESCRIPTION
**Pull request by tomkralidis**
- support CRS in collections
- implement temporal in collections
- handle `items` datetime queries
- refactor OpenAPI responses as reusable components
- add JSON datetime helper function
- update collection.name to collection.id
- adds `itemType` to collections response

TODO:
- [x] add tests for temporal
- [x] sort out temporal extent types for `indeterminatePosition="now"` pattern
- [x] try to link more content model YAML schemas in OpenAPI doc
- [x] update `time` to `datetime` for `items` queries

### CODE REVIEW

**Comment by tomkralidis** [comment]
@francbartoli this PR is now ready for review.

File: pygeoapi/openapi.py [hunk 1]
```diff
@@ -107,9 +155,9 @@ def get_oas_30(cfg):
                 {'$ref': '#/components/parameters/f'}
             ],
             'responses': {
-                200: {
-                    'description': 'successful operation'
-                }
+                200: {'$ref': '#/components/responses/200'},
+                400: {'$ref': '{}#/components/responses/InvalidParameter'.format(OPENAPI_YAML['oapif'])},  # noqa
```
[the remark this answers is not in the document]
Comment (tomkralidis) [reply pygeoapi/openapi.py]: fixed.

**Comment by justb4** [comment]
Ok, demo fixed. Suggestion: make `crs` (like `trs`) optional with default of http://www.opengis.net/def/crs/OGC/1.3/CRS84 .

**Comment by justb4** [comment]
Interestingly the GHC WFS3 Probe still fails:
https://demo.geohealthcheck.org/resource/165
Has to do with `name` attr not found in `collection` (changed to `id`?)

**Comment by tomkralidis** [comment]
Yes. collection.name has changed to collection.id.

**Pull request merged**